waiverdb/api_v1.py

```diff
@@ -356,24 +246,25 @@ def post(self):
         user, headers = waiverdb.auth.get_user(request)
         data = request.get_json(force=True)
 
-        if isinstance(data, list):
-            result = []
-            for sub_data in data:
-                sub_request = DummyJsonRequest(sub_data)
-                args = RP['create_waiver'].parse_args(sub_request)
-                one_result = self._create_waiver(args, user)
-                result.append(one_result)
-            db.session.add_all(result)
-        else:
-            args = RP['create_waiver'].parse_args()
-            result = self._create_waiver(args, user)
-            db.session.add(result)
+        try:
+            if isinstance(data, list):
+                result = [
+                    self._create_waiver(CreateWaiver.parse_obj(sub_data), user)
+                    for sub_data in data
+                ]
+                db.session.add_all(result)
+            else:
+                result = self._create_waiver(CreateWaiver.parse_obj(data), user)
+                db.session.add(result)
+        except pydantic.ValidationError as e:
+            raise BadRequest(parse_validation_error(e)) from e
```

Comment:
This is handled by `@validate` decorator from `Flask-Pydantic`. Not sure how well it would work on the methods in Resource.

```
    @jsonp
    @marshal_with(waiver_fields)
    @validate()
    def post(self, body: CreateWaiver):
        ...
```